src/Components/ADT3DSceneBuilder/Internal/VisualRuleForm/VisualRuleForm.tsx

```diff
@@ -258,6 +283,24 @@ const VisualRuleForm: React.FC<IVisualRuleFormProps> = (props) => {
         [visualRuleFormDispatch]
     );
 
+    const handleSaveCondition = useCallback(
+        (condition: IValueRange) => {
+            visualRuleFormDispatch({
+                type: VisualRuleFormActionType.FORM_CONDITION_ADD_OR_UPDATE,
+                payload: {
+                    condition: {
+                        ...condition,
+                        visual: {
+                            ...condition.visual,
+                            labelExpression: condition.visual.labelExpression.trim()
```

Comment:
I see we have to do this trimming again on save, why do we not just store the valid value so we don't have to do these manipulations and all the object expansions?

Reply:
See answers on trimming above.